Agents/AssetManagerAgent/AssetManagerAgent/src/test/java/uk/ac/cam/cares/jps/agent/assetmanager/AssetManagerAgentTest.java

```diff
@@ -0,0 +1,642 @@
+        clientPropParam.add("url.manual=https://www.theworldavatar.com:1010/careslab/manuals/");
+        clientPropParam.add("url.docupload=http://host.docker.internal:1016/document-upload-agent/adddocument");
+
+        String[] ontoMapPropParam = {"LabEquipment=https://www.theworldavatar.com/kg/ontoassetmanagement/LabEquipment",
+            "Laptop=https://www.theworldavatar.com/kg/ontoassetmanagement/Laptop",
+            "Monitor=https://www.theworldavatar.com/kg/ontoassetmanagement/Monitor",
+            "OtherIT=https://www.theworldavatar.com/kg/ontoassetmanagement/OtherIT",
+            "Printer=https://www.theworldavatar.com/kg/ontoassetmanagement/Printer",
+            "Workstation=https://www.theworldavatar.com/kg/ontoassetmanagement/Workstation"};
+        String[] tsSearchPropParam = {"depth =" + String.valueOf(5),
+            "predicate=https://www.theworldavatar.com/kg/ontotimeseries/hasTimeSeries,"+
+             "https://saref.etsi.org/core/hasState,"+
+             "https://www.theworldavatar.com/kg/ontobms/hasLowLevelState,"+
+             "https://www.theworldavatar.com/kg/ontobms/hasMediumLevelState,"+
+             "https://www.theworldavatar.com/kg/ontobms/hasHighLevelState,"+
+             "https://www.theworldavatar.com/kg/ontodevice/sendsSignalTo,"+
+             "https://saref.etsi.org/core/consistsOf,"+
+             "https://www.theworldavatar.com/kg/ontodevice/measures,"+
+             "http://www.ontology-of-units-of-measure.org/resource/om-2/hasValue,"+
+             "https://www.theworldavatar.com/kg/ontodevice/observes" };
+        writePropertyFile(agentPropFile, clientPropParam);
+        writePropertyFile(ontoMapPropFile, Arrays.asList(ontoMapPropParam));
+        writePropertyFile(tsSearchPropFile, Arrays.asList(tsSearchPropParam));
+
+        //Set the RemoteStoreClient
+        for (Map.Entry<String, String> set  : namespaceMap.entrySet()){
+            String ns = set.getKey();
+            String endpoint = set.getValue();
+            storeClientMap.put(ns, new RemoteStoreClient(endpoint, endpoint));
+        }
+        
+        // To create testAgent without an exception being thrown, SystemLambda is used to mock an environment variable
+        // To mock the environment variable, a try catch need to be used
+        /*ENV AGENTPROPERTIES "/root/agent.properties"
+        ENV ONTOMAPPROPERTIES "/root/ontologyMap.properties"
+        ENV TSSEARCH "/root/tsSearch.properties"
+        ENV FOLDERQR "/root/folderQR/"
+        ENV FOLDERMANUAL "/root/manuals/"*/
+        try {
+        	SystemLambda.withEnvironmentVariable("AGENTPROPERTIES", agentPropFile)
+                .and("ONTOMAPPROPERTIES", ontoMapPropFile)
+                .and("TSSEARCH", tsSearchPropFile)
+                .and("FOLDERQR", folderQRFilepath)
+                .and("FOLDERMANUAL", folderManualFilepath)
+                .execute(() -> {
+                agent = new AssetManagerAgent();
```

Comment:
the environment variables only work in this block, and not found in other test cases (eg. `TestRetrieveUIData()`). need to fix this